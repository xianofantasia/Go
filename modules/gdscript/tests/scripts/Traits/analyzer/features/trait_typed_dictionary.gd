trait SomeTrait:
	pass

class SomeClass:
	uses SomeTrait

func test():
	var key_classes: Dictionary[SomeTrait, String] = {SomeClass.new() : "A"}
	key_classes[SomeClass.new()] = "B"
	var count = 1
	for trait_class in key_classes:
		print(count)
		count += 1
	print("key ok")
	var value_classes: Dictionary[String, SomeTrait] = {"A" : SomeClass.new()}
	value_classes["B"] = SomeClass.new()
	count = 1
	for trait_class in value_classes:
		print(count)
		count += 1
	print("value ok")
