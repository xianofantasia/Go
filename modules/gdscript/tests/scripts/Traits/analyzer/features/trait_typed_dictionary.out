GDTEST_OK
1
2
key ok
1
2
value ok
